internal/subscriber/core_test.go

```diff
@@ -0,0 +1,364 @@
+	subscrption := getDummySubscriptionWithRetry()
+	type fields struct {
+		core *Core
+	}
+	type args struct {
+		ctx                    context.Context
+		subscriberID           string
+		subscription           *subscription.Model
+		timeoutInMs            int
+		maxOutstandingMessages int64
+		maxOutstandingBytes    int64
+		requestCh              chan *PullRequest
+		ackCh                  chan *AckMessage
+		modAckCh               chan *ModAckMessage
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		expected    *Subscriber
+		expectedMsg []string
+		wantErr     bool
+	}{
+		{
+			name: "Create new subscriber with retry policy",
+			fields: fields{
+				core: getCore(store, core, offsetCore, ch),
+			},
+			args: args{
+				ctx:                    ctx,
+				subscriberID:           subID,
+				subscription:           &subscrption,
+				timeoutInMs:            1000,
+				maxOutstandingMessages: 1000,
+				maxOutstandingBytes:    1000,
+				requestCh:              requestCh,
+				ackCh:                  ackCh,
+				modAckCh:               modAckCh,
+			},
+			expected: &Subscriber{
+				subscription: &subscrption,
+				topic:        topicName,
+				subscriberID: subID,
+				requestChan:  requestCh,
+				ackChan:      ackCh,
+				modAckChan:   modAckCh,
+			},
+			expectedMsg: []string{"a", "b"},
+			wantErr:     false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			store.EXPECT().GetConsumer(gomock.Any(), gomock.Any()).
+				DoAndReturn(func(arg0 context.Context, arg1 messagebroker.ConsumerClientOptions) (messagebroker.Consumer, error) {
+					if tt.wantErr {
+						return nil, errors.New("Test Error")
+					}
+					return cs, nil
+				}).AnyTimes()
+			count := 0
+			cs.EXPECT().ReceiveMessages(gomock.Any(), gomock.Any()).DoAndReturn(
+				func(arg0 context.Context, arg1 messagebroker.GetMessagesFromTopicRequest) (*messagebroker.GetMessagesFromTopicResponse, error) {
+					if count == 0 {
+						count++
+						return &messagebroker.GetMessagesFromTopicResponse{Messages: getMockReceivedMessages(tt.expectedMsg)}, nil
+					}
+					return &messagebroker.GetMessagesFromTopicResponse{}, nil
+				}).AnyTimes()
+			broker.EXPECT().Resume(gomock.Any(), gomock.Any()).AnyTimes().Return(nil)
+			cs.EXPECT().Close(gomock.Any()).Return(nil).AnyTimes()
+			store.EXPECT().RemoveConsumer(gomock.Any(), gomock.Any()).Return(true).AnyTimes()
+			sub, err := tt.fields.core.NewSubscriber(tt.args.ctx, tt.args.subscriberID, tt.args.subscription, tt.args.timeoutInMs,
+				tt.args.maxOutstandingMessages, tt.args.maxOutstandingBytes, tt.args.requestCh, tt.args.ackCh, tt.args.modAckCh)
+			if err == nil {
+				assert.True(t, EqualOnly(sub, tt.expected, []string{"topic", "subscriberID", "requestChan", "ackChan", "modAckChan"}))
+			}
+			if (err != nil && !tt.wantErr) || (err == nil && tt.wantErr) {
+				t.Errorf("NewSubscriber wantErr: %v Error: %v", tt.wantErr, err)
+			}
+			sub.Stop()
```

Comment:
Here we have a subscriptions with retry policy just checking if we did not get any error while running subscriber with retry. Since i do not have any access to subImpl i could not assert anything.

Reply:
Then it's not a valid test case. You can merge the above two test cases.

Reply:
merged two test cases